lib/instana/instrumentation/instrumented_request.rb

```diff
@@ -9,7 +9,7 @@
 
 module Instana
   class InstrumentedRequest < Rack::Request
-    W3_TRACE_PARENT_FORMAT = /00-(?<trace>[0-9a-f]+)-(?<parent>[0-9a-f]+)-(?<flags>[0-9a-f]+)/.freeze
+    W3C_TRACE_PARENT_FORMAT = /[0-9a-f][0-9a-e]-(?<trace>[0-9a-f]{32})-(?<parent>[0-9a-f]{16})-(?<flags>[0-9a-f]{2})(?:-.*)?/.freeze
```

Comment:
The trailing non-capturing group seems a bit superfluous to me.
We didn't add it in the python tracer, and we don't seem to need it here either.

Empirically, the matching of theoretical future formats with additional fields, is just as good without it:
````
❯ irb
irb(main):001:0> W3_TRACE_PARENT_FORMAT = /[0-9a-f][0-9a-e]-(?<trace>[0-9a-f]{32})-(?<parent>[0-9a-f]{16})-(?<flags>[0-9a-f]{2})/.freeze
=> /[0-9a-f][0-9a-e]-(?<trace>[0-9a-f]{32})-(?<parent>[0-9a-f]{16})-(?<flags>[0-9a-f]{2})/
irb(main):002:0> 'fe-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01-abcdefg'.match(W3_TRACE_PARENT_FORMAT)
=> #<MatchData "fe-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01" trace:"4bf92f3577b34da6a3ce929d0e0e4736" parent:"00f067aa0ba902b7" flags:"01">
irb(main):003:0> 
````
Unless there is a strong reason to increase the complexity of the regex, I would vote on applying Occam's razor here.

Reply:
You are absolutely right. I was somehow under the impression that I need to consume the entire string with the regex to produce a match, but it is actually a substring match. Fixed.